src/TargetResourceHandler.cpp

```diff
@@ -0,0 +1,115 @@
+#include "TargetResourceHandler.hpp"
+
+/**
+ * @brief Construct a new TargetResourceHandler object
+ *
+ * @param locations Vector of Location objects of current server block.
+ * @param request Request to handle.
+ * @param response Response to be filled.
+ * @param fileSystemPolicy File system policy to be used.
+ */
+TargetResourceHandler::TargetResourceHandler(const std::vector<Location>& locations, const HTTPRequest& request,
+	HTTPResponse& response, const FileSystemPolicy& fileSystemPolicy)
+	: m_locations(locations)
+	, m_request(request)
+	, m_response(response)
+	, m_fileSystemPolicy(fileSystemPolicy)
+{
+}
+
+/**
+ * @brief Determine target resource based on the request path and handle different file types and statuses accordingly.
+ *
+ * Finds a matching location block.
+ * If no match is found set Status to StatusNotFound.
+ * Set the targetResource as location.root + request.uri.path and check file type.
+ * RegularFile: break, nothing more to do.
+ * Directory: if no trailing slash, add it and set Status to StatusMovedPermanently.
+ *            if index file is set, add it to the path and set internalRedirect to true.
+ *            if autoindex is set, set autoindex to true.
+ *            else set Status to StatusForbidden.
+ * FileNotExist: set Status to StatusNotFound.
+ * FileOther: set Status to StatusForbidden.
+ * If internalRedirect is true, repeat the process.
+ * If stat fails, set Status to StatusInternalServerError.
+ */
+void TargetResourceHandler::execute()
+{
+	bool internalRedirect = false;
+
+	do {
+		// Check which location block matches the path
+		m_response.location = matchLocation(m_request.uri.path);
+
+		// No location found > do we also set a default location to not make extra check?
```

Comment:
In nginx if you don't define a location in a server config you get a default location which matches "/" (i.e. everything)
If you set a location "/test", then you don't have a "/" location.

We could implement this by creating a default location "/" which I can also loop through (so I dont need an extra check: "is there a default location?")

The behaviour you're describing is handled in the next three lines below: if we don't find a matching location we set StatusNotFound. This would create the error response you're describing later via in the ResponseBodyHandler and the appendStatusLine() and appendHeaders().

Reply:
To get on the same page:
We are discussing the situation that the requested target resource location is not found among any of the locations in our server (defined by the config file).

In that case you want to go to a default location like "/" and look for the target resource there? Despite the requested location being another?

If yes, the question begs: is that extra step really necessary? Instead of directly defaulting to creating the "not found" answer.

Reply:
> We are discussing the situation that the requested target resource location is not found among any of the locations in our server (defined by the config file).

Yes.

> In that case you want to go to a default location like "/" and look for the target resource there? Despite the requested location being another?

Its more of an implementation detail: Say you have only one location block "/secret" and you request "/something". It would match nothing. 

We could then just report Not found and leave (this is currently implemented). This means we have to always explicitly define all locations.

We could also have a default location "/" which is always implicitly there. So the ConfigFileParser would just init this location. Since a HTTP request has to have a slash, this location would be the catch-all for all requests.
We would still search for /something but in the default location. 

Consider nginx: when you have a very plain config:
```
http {
  server {
    root /data/www;
  }
} 
```

nginx will serve you, when you request "/" --> it will give you /data/www/index.html. Even though we didn't specify any location.
Even if you do specify one location like
```
http {
  server {
    root /data/www;
    location /secret {
    }
  }
} 
```
nginx will still serve you, when you request "/" --> it will give you /data/www/index.html.
It will also serve you, when you request "/something" --> it will give you /data/www/something/index.html

I assume the location for "/" is always defined implicitly. (also: index.html seems to be also implicitly defined )

What doesn't work for me is:
```
http {
  server {
  }
} 
```
-> no root specified. But it still is a valid config. I suspect that the default root, which should be "html" was changed and not updated in manual.

Reply:
> It will also serve you, when you request "/something" --> it will give you /data/www/something/index.html

Even if we don't have a folder "something" inside of /data/www/ ?

Reply:
The target would be /data/www/something/index.html. 
It gets checked in the following try block at line 56 ff
If there is no such file then it will 404 File not found.

If we don't want to go with the default "/" location we could exit early (this is impl right now)
If we implement the default location we would of course need to check if the file exists (in the following lines)

Reply:
After discussion agreed on integration of default location.